### PR TITLE
Allow duplicate keys for most mappings except stacks.

### PR DESCRIPTION
**Pull request by russellballestrini**
modified:   stacker/tests/test_config.py
	modified:   stacker/util.py

### CODE REVIEW

**Comment by acmcelwee** [comment]
Is this to allow defaults with overrides or something along those lines? If so, what about warn on dupes if they're not in the hard fail set of keys?

**Comment by russellballestrini** [comment]
Yeah, I prefer my original attempt to fix this issue.

It was found that some end users may be setting a default for a some keys and then redefine the keys later on to override.

https://github.com/remind101/stacker/pull/500

The YAML spec is sort of confusing. It seems to indicate that keys "MUST" be unique in some places.

```
JSON's RFC4627 requires that mappings keys merely “SHOULD” be unique, while YAML insists they “MUST” be. Technically, YAML therefore complies with the JSON spec, choosing to treat duplicates as an error. In practice, since JSON is silent on the semantics of such duplicates, the only portable JSON files are those with unique keys, which are therefore valid YAML files.
```

Reference:  http://www.yaml.org/spec/1.2/spec.html section 1.3. Relation to JSON

File: stacker/tests/test_config.py [hunk 1]
```diff
@@ -408,10 +408,21 @@ def test_raise_constructor_error_on_duplicate_key(self):
             class_path: blueprints.VPC
             class_path: blueprints.Fake
         """
-        with self.assertRaises(ConstructorError):
-            parse(yaml_config)
+        # with self.assertRaises(ConstructorError):
```
Comment (phobologic) [inline stacker/tests/test_config.py]: Can you remove these commented out lines?

Comment (phobologic) [reply stacker/tests/test_config.py]: Also, it seems like this test is named incorrectly.

**Comment by phobologic** [comment]
Thanks @russellballestrini

**Pull request merged**